### ISSUE TITLE
Deprecation warning ember-getowner-polyfill.import

### ISSUE THREAD

**Issue by janmisek** [rwjblue/ember-getowner-polyfill#15]
Hello 

I am getting deprecation warning `ember-getowner-polyfill.import` caused by addons using the polyfill as caret dependency `^`. I am currently on ember 2.9.1 and I like to keep app clean of deprecations. Even because of clean console. Is there any solution for this?

I am now using following workaround after npm install:
```
find node_modules -type d -name ember-getowner-polyfill | xargs rm -fr
npm install ember-getowner-polyfill@1.0.1
```

Thanks in advance

**Comment by janmisek** [comment]
Currently I am using following workaround to get rid of deprecation warning. Its part of my install pipe and basically it installs everything then remove latest `ember-getowner-polyfill` and force installation of 1.0.1 version which does not include deprecation warning.

```
find node_modules -type d -name ember-getowner-polyfill | xargs rm -fr
npm install ember-getowner-polyfill@1.0.1
```

**Comment by Dhaulagiri** [comment]
@janmisek you may already be familiar with it, but I really like using [ember-cli-deprecation-workflow](https://github.com/mixonic/ember-cli-deprecation-workflow) for these situations where I can't get rid of a deprecation due to upstream issues in addons but I also don't want to see it in the console

**Comment by rwjblue** [comment]
Indeed, thanks for chiming in here!

**Comment by Turbo87** [comment]
it should also be mentioned that the best way to resolve this is opening PRs on the addons and resolving the deprecation issue!

**Comment by rwjblue** [comment]
Confirm

**Comment by janmisek** [comment]
I agree and I also like deprecation workflow. But I think in this particular case versioning is responsible. Like deprecation is upcoming bc therefore deprecation should not be considered as minor change considering that most of the addons link the module as ~

**Comment by bluscreen** [comment]
I'm also suffering from this warning as well as htmlsafe-polyfill

```
DEPRECATION: ember-string-ishtmlsafe-polyfill is now a true polyfill. Use Ember.String.isHTMLSafe directly instead of importing from ember-string-ishtmlsafe-polyfill [deprecation id: ember-string-ishtmlsafe-polyfill.import]

DEPRECATION: ember-getowner-polyfill is now a true polyfill. Use Ember.getOwner directly instead of importing from ember-getowner-polyfill [deprecation id: ember-getowner-polyfill.import]
```

does anybody know how to actually cure them? i cant find any direct references in my project for including them...

**Comment by workmanw** [comment]
@bluscreen I suspect you're using an addon that is using the polyfill. I'm not sure if it's possible, but we should only throw this deprecation if the app is directly using it. Or perhaps we should not even throw the deprecation at all.

**Comment by bluscreen** [comment]
i personally would say that it's not really such a critical thing to still have them included in some plugins..

**Comment by bcardarella** [comment]
> it should also be mentioned that the best way to resolve this is opening PRs on the addons and resolving the deprecation issue!

@Turbo87 the problem with this is that some of the more popular addons are not in any hurry to move away from the polyfill. They want to support the entire 2.x major version cycle.

**Comment by Turbo87** [comment]
@bcardarella I wasn't suggesting to remove the polyfill, just to update it to the latest version where you don't have to import it explicitly which creates the deprecation message above.

**Comment by rwjblue** [comment]
Yes, the current version of the polyfill is a true polyfill. You use the exact same API that you would use on newer Ember versions.

**Comment by Turbo87** [comment]
@bcardarella I suppose what you're actually commenting on is the situation described in #16, right?

**Comment by bcardarella** [comment]
Its more about the necessity for this particular addon to put the burden of the warning upon the application consumer. Whereas it is really meant for the addon developer that directly consumes it. The app developer has little recourse for resolving the warning if the addon developer is unable or unwilling to upgrade or resolve. Yet they are the one who constantly has to see the message.

**Comment by bcardarella** [comment]
The irony however, is that even if the warning was removed or somehow made so that it only shows while developing the addon, there would still need to be an update made which puts us back at square one. So I'm hoping that in the future when it comes to something like this in Ember there can be an easier course of action to not impact app developers as much

**Comment by bcardarella** [comment]
Also, I think it would be helpful if somehow deprecation warnings could notify which addon is the one offending. Doing `npm ls` to search for the dependency doesn't give reliable results on which addon depends on `ember-getowner-polyfill`. Considering how many addons use it, and tracking down which ones are not using it correctly, is difficult.

Perhaps this is something that the EmberObserver folks can help with? If they're already doing some sort of static analysis on each addon perhaps they can help track down the ones that are still importing this addon and we can try and push those addon developers to updating?

**Comment by rwjblue** [comment]
EmberObserver has code search built in, there are 49 addons that have `from 'ember-getowner-polyfill'` in them ([see here](https://emberobserver.com/code-search?codeQuery=from%20%27ember-getowner-polyfill%27)).

**Comment by rwjblue** [comment]
> Also, I think it would be helpful if somehow deprecation warnings could notify which addon is the one offending.

FWIW, there is no way to know the "referrer" of an import.  It is possible to add that (https://github.com/ember-cli/loader.js/pull/118 makes it a bit easier), but it doesn't exist today.

**Comment by rwjblue** [comment]
The deprecation has been removed and published in 2.0.0.
